src/main/java/net/dries007/tfc/objects/items/metal/ItemAnvil.java

```diff
@@ -62,7 +62,9 @@ public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos p
         {
             ItemStack stack = player.getHeldItem(hand);
             IBlockState state = worldIn.getBlockState(pos.offset(facing));
-            if (state.getBlock().isReplaceable(worldIn, pos.offset(facing)))
+            IBlockState stateDown = worldIn.getBlockState(pos.offset(facing).down());
+            if (state.getBlock().isReplaceable(worldIn, pos.offset(facing)) &&
+                stateDown.isSideSolid(worldIn, pos.offset(facing).down(), EnumFacing.UP)) //forge says to do it this way, IBlockProperties#isTopSolid
```

Comment:
You're offsetting from an offsetted IBlockState already, and again offsetting by calling .down() on the BlockPos.

Reply:
I've cleaned this up. I'll claim I only added two .down() calls to this mess, but the rest were already there. ;)

Thanks for the review!